cpu/efm32/periph/adc_series2.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2022 SSV Software Systems GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_efm32
+ * @ingroup     drivers_periph_adc
+ * @{
+ *
+ * @file
+ * @brief       Low-level ADC driver implementation
+ *
+ * @author      Juergen Fitschen <me@jue.yt>
+ *
+ * @}
+ */
+
+#include <assert.h>
+#include <string.h>
+
+#include "cpu.h"
+#include "mutex.h"
+
+#include "periph_conf.h"
+#include "periph/adc.h"
+#include "periph/gpio.h"
+
+#include "em_cmu.h"
+#include "em_gpio.h"
+#include "em_iadc.h"
+
+static mutex_t adc_lock[ADC_DEV_NUMOF];
+
+int adc_init(adc_t line)
+{
+    /* check if line is valid */
+    if (line >= ADC_NUMOF) {
+        return -1;
+    }
+
+    uint8_t dev = adc_channel_config[line].dev;
+    assert(dev < ADC_DEV_NUMOF);
+
+    /* initialize lock */
+    mutex_init(&adc_lock[dev]);
+
+    /* enable clock */
+    CMU_ClockEnable(adc_config[dev].cmu, true);
+
+    /* make sure we're in a known state */
+    IADC_reset(adc_config[dev].dev);
+
+    /* init IADC periph */
+    const IADC_Init_t init = IADC_INIT_DEFAULT;
+    IADC_AllConfigs_t configs;
+    memset(&configs, 0x00, sizeof(configs));
```

Comment:
Nitpick, but could we do

```suggestion
    IADC_AllConfigs_t configs = { 0 };
```

instead?

Reply:
Hmmm, I learned in the past that this is guaranteed to work ... but I guess that's an old C-standard? But sure if this works, this is the better and way more expressive solution to zero this struct.

Reply:
85923524ff6ba9b1fc525b7cf3cf0aed8446df08